docs/IE_DG/ShapeInference.md

```diff
@@ -1,38 +1,60 @@
+for (const auto & parameter : parameters) {
+    std::cout << "name: " << parameter->get_friendly_name() << " shape: " << parameter->get_partial_shape() << std::endl;
+    if (parameter->get_partial_shape().is_dynamic())
+        std::cout << "ATTENTION: input shape is not fully defined, please use CNNNetwork::reshape method to resolve it!" << std::endl;
+}
+```
+
+To feed input data of a shape that is different from the model input shape, reshape the model first.
+
+OpenVINO™ provides the following methods for runtime model reshaping:
+
+* **Set a new input shape** with the `InferenceEngine::CNNNetwork::reshape` method.<br>
+   The `InferenceEngine::CNNNetwork::reshape` method updates input shapes and propagates them down to the outputs of the model through all intermediate layers. It is allowed to reshape model multiple times like in this application scheme:
+   ```
+   ReadNetwork -> reshape(input_1_shape) -> LoadNetwork -> infer(input_1)
+              \
+               -> reshape(input_2_shape) -> LoadNetwork -> infer(input_2)
+   ```
+   > **NOTES**:
+   > - Starting with the 2021.1 release, the Model Optimizer converts topologies keeping shape-calculating sub-graphs by default, which allows correct shape propagation during    reshaping.<br>
+   > - Older versions of IRs are not guaranteed to reshape successfully. Please re-generate them with the Model Optimizer of the latest version of OpenVINO™.<br>
+   > - It is required to reshape the network before loading to the plugin for ONNX models with not fully defined input shape that were imported with the ONNX importer.
+* **Set a new batch dimension value** with the `InferenceEngine::CNNNetwork::setBatchSize` method.<br>     
+   The meaning of a model batch may vary depending on choices you made during the model designing. 
+   The `InferenceEngine::CNNNetwork::setBatchSize` method deduces index of batch dimension relying only on the input rank. 
+   This method does not work for models with a non-zero index batch placement or models with inputs without a batch dimension. 
+   Batch-setting algorithm does not involve shape inference mechanism.
+   Batch of input and output shapes for all layers is set to a new batch value without layer validation.
+   It may cause both positive and negative side effects.
+   Due to the limitations described above, the current method is not recommended to use.
+   If there is a need to set new batch size for the model, please use `CNNNetwork::reshape` method instead.
+
+Please do not use runtime reshaping methods together, especially do not call `CNNNetwork::reshape` method after `InferenceEngine::CNNNetwork::setBatchSize` usage.
+`InferenceEngine::CNNNetwork::setBatchSize` method causes irreversible conversion of internal model representation to the legacy model representation.
+It does not use `nGraph` for shape inference which lead to reduced reshape opportunities and may affect performance of the model.
```

Comment:
The method does not use `nGraph` for shape inference which leads to reduced reshape opportunities and may affect the performance of the model.